gen3-integration-tests/services/sheepdog.py

```diff
@@ -0,0 +1,154 @@
+import copy
+import json
+import os
+import uuid
+
+from cdislogging import get_logger
+from gen3.submission import Gen3Submission
+import pytest
+import requests
+
+from utils import TEST_DATA_PATH
+
+
+logger = get_logger(__name__, log_level=os.getenv("LOG_LEVEL", "info"))
+
+
+class GraphRecord:
+    def __init__(self, node_name, category, submission_order, props):
+        """
+        node_name: id of the node in the dictionary (example: "case")
+        category: category of the node in the dictionary (example: "administrative")
+        submission_order: index of this record in the submission order (example: 2)
+        props: record as it should be submitted to sheepdog
+        sheepdog_id: unique "id" property returned by sheepdog when submitting a record
+        indexd_guid: for *_file nodes, the guid of the referenced file in indexd
+        """
+        self.node_name = node_name
+        # self.category = category  # not used yet
+        self.submission_order = submission_order
+        self.props = props
+        self.sheepdog_id = None
+        # self.indexd_guid = None  # not used yet
+
+    def __str__(self):
+        return f"GraphRecord '{self.node_name}': {self.props}"
+
+
+class Sheepdog:
+    def __init__(self, auth, program_name="jnkns", project_code="jenkins"):
+        self.sdk = Gen3Submission(auth_provider=auth)  # endpoint=pytest.root_url
+        self.program_name = program_name
+        self.project_code = project_code
+        self.test_data_path = TEST_DATA_PATH / "graph_data"
+        self.submission_order = []  # node names in the order they should be submitted
+        # records as generated by data-simulator - { node name: GraphRecord }
+        self.test_records = {}
+        self.load_test_records()
+
+    def create_program_and_project(self):
```

Comment:
We should probably follow our same coding practices for non-test code and require docstrings on public functions